pkg/adaptor/elasticsearch/transport.go

```diff
@@ -0,0 +1,33 @@
+package elasticsearch
+
+import (
+	"net/http"
+
+	awsauth "github.com/smartystreets/go-aws-auth"
+)
+
+// AWSTransport handles wrapping requests to AWS Elasticsearch service
+type AWSTransport struct {
+	Credentials awsauth.Credentials
+	transport   http.RoundTripper
+}
+
+func newTransport(accessKeyID, secretAccessKey string) http.RoundTripper {
+	t := http.DefaultTransport
+	if accessKeyID != "" && secretAccessKey != "" {
```

Comment:
what would we be testing here? when adding support for AWS signed auth, this seemed like the best/cleanest route to go.